kiosk/src/Components/ScanQR.tsx

```diff
@@ -26,11 +26,16 @@ const ScanQR: React.FC<IProps> = ({ kiosk }) => {
     const checkUrl = async () => {
         const input = document.getElementById("kiosk-share-link") as HTMLInputElement;
         const inputValue = input.value;
-        const shareLink = /(https:\/\/)((arcade\.makecode\.com\/)|(makecode\.com\/))((?:S\d{5}-\d{5}-\d{5}-\d{5})|(?:_[a-zA-Z0-9]+))/.exec(inputValue);
+        const shareLink = /^(https:\/\/)((arcade\.makecode\.com\/)|(makecode\.com\/))((?:S\d{5}-\d{5}-\d{5}-\d{5})$|(?:_[a-zA-Z0-9]+)$)/.exec(inputValue);
```

Comment:
```suggestion
        const shareLink = /^(https:\/\/)((arcade\.makecode\.com\/)|(makecode\.com\/))((?:S?\d{5}-\d{5}-\d{5}-\d{5})$|(?:_[a-zA-Z0-9]+)$)/i.exec(inputValue);
```

This isn't handling normal / normal non persistent share links like https://arcade.makecode.com/03212-84866-58948-31655 (and added /i since domain names are case insensitive and it's possible a user might type it in wrong. technically that would allow 
https://arcade.makecode.com/s32800-39924-12945-26540 but we have to check the share id later on anyways so no real harm)

Reply:
Thank you for pointing that out! I thought that non-persistent share links always looked like
https://makecode.com/_f0jhEb99UFkP since this is what is generated in the share dialog when you uncheck update the game with the changes.